### PR TITLE
Add hidden savefile backup and automatic export

### LINKED ISSUE #36: Should we leave `Export` as it is?
The `Export` command is only performed if the user specifies it.

There are some Pros and Cons to this:

Pros
- The user can decide when to export the card list after modifying it as much as they want.

Cons
- The user must remember to export the document each time, and the automatic load and save component is lost. @ltzehan suggested that we export before exiting the program every time; this seems fair.

### LINKED ISSUE #46: Corrupted Json files crash the program
Caused by: com.google.gson.stream.MalformedJsonException

However, the corrupted save file warning is still triggered, so the error is still thrown

### PR DESCRIPTION
**Pull request by iantenkai**
1. fixes #46 :
Sets the reader to lenient

2. closes #36:
Exports the deck automatically on exit 
Creates a hidden backup on save.
Loads the backup if the original is corrupted.

### CODE REVIEW

File: src/main/java/utils/command/TerminateCommand.java [hunk 1]
```diff
@@ -10,6 +10,7 @@ public class TerminateCommand extends Command {
     @Override
     public void execute(CardList cardList, TagList tagList, UserInterface ui, IDataStorage storage)
             throws InkaException {
+        storage.save(cardList);
```
Comment (ltzehan) [inline src/main/java/utils/command/TerminateCommand.java:13]: This makes sense, but what if the user exits the program abruptly (e.g., Ctrl-C or closing the window)

Should we just save this after every command that modifies the cards?

File: src/main/java/utils/storage/json/JsonStorage.java [hunk 2]
```diff
@@ -63,13 +239,35 @@ public CardList load() throws InkaException {
         } catch (NullPointerException | JsonSyntaxException e) {
             String absolutePath = this.saveFile.getAbsolutePath();
             logger.log(Level.WARNING, "Corrupted save file: " + absolutePath, e);
-
+            useBackup = true;
             throw new StorageCorrupted(absolutePath);
         }
 
+        if (useBackup == true) {
```
Comment (ltzehan) [inline src/main/java/utils/storage/json/JsonStorage.java:246]: Can rewrite this as `if (useBackup)`

File: src/main/java/utils/storage/json/JsonStorage.java [hunk 3]
```diff
@@ -63,13 +239,35 @@ public CardList load() throws InkaException {
...
+            logger.log(Level.INFO, "Trying to load backup file");
+            try{  FileReader fileReader = new FileReader(backupFile);
```
Comment (ltzehan) [inline src/main/java/utils/storage/json/JsonStorage.java:248]: Can replace this with a [try-with-resource](https://www.baeldung.com/java-try-with-resources) so you don't have to explictly close `fileReader` or `bufferedReader`

File: src/main/java/utils/storage/json/JsonStorage.java [hunk 4]
```diff
@@ -1,6 +1,169 @@
+
```
Comment (ltzehan) [inline src/main/java/utils/storage/json/JsonStorage.java:1]: 👀

File: src/main/java/utils/storage/json/JsonStorage.java [hunk 5]
```diff
@@ -29,32 +191,46 @@
 public class JsonStorage extends Storage {
     private static Logger logger = Logger.getLogger("storage.JsonStorage");
     private GsonBuilder gsonBuilder;
+    private File backupFile;
 
     public JsonStorage(String filePath) {
         super(filePath);
+
+        // Create the backup file
+        String backupFilePath = filePath.replace(".json", "_backup.json");
```
Comment (ltzehan) [inline src/main/java/utils/storage/json/JsonStorage.java:200]: Could pull this out into `private File getBackupFile()` too

**Review by lihka1202 — APPROVED** [review]
Approved by @lihka1202

**Pull request merged**